### ISSUE TITLE
[BUG] Media-query-order wrapping diverges from Babel when a max-width-only query is present ("(not all) or" chains vs authored queries)

### ISSUE THREAD

**Issue by matclayton** [Dwlad90/stylex-swc-plugin#1268]
### Describe the bug

When a conditional value map contains a `max-width`-only query alongside exclusive `min-width`/`max-width` ranges, `@stylexjs/babel-plugin` wraps the earlier entries in `(not all) or …` chains; this compiler emits the authored queries unchanged. The outputs are semantically equivalent (`(not all) or Q` ≡ `Q`), but the rule text and class hashes differ, so the two compilers cannot be mixed.

To be explicit about which side we think is "right": the Rust output is cleaner, and Babel's wrapper looks like a Babel bug — it is redundant, and its double-parenthesised form (`(not all) or ((min-width: 1632px))`) is rejected by lightningcss's minifier (`Unexpected token ParenthesisBlock`). We are reporting the divergence here for hash-parity completeness and cross-reporting the wrapper itself to facebook/stylex; matching Babel byte-for-byte versus documenting a deliberate divergence is your call.

### Error stack

No error — the transform succeeds but emitted CSS and class hashes differ from `@stylexjs/babel-plugin`. Outputs below.

### OS

Linux x64 (also reproduced on macOS arm64)

### Browser

n/a — compile-time divergence

### Compiler version

@stylexswc/rs-compiler 0.18.4-rc.2 (also present on 0.18.3)

### Bundler plugin(s) version

n/a — `transform()` from `@stylexswc/rs-compiler` called directly (no bundler in the loop)

### To Reproduce

1. Install `@stylexswc/rs-compiler@0.18.4-rc.2` and `@stylexjs/babel-plugin@0.19.0`.
2. Run the input below through both compilers with identical options: `{ dev: false, treeshakeCompensation: true, unstable_moduleResolution: { type: 'commonJS', rootDir: process.cwd() } }` — for the Babel side via `transformAsync` with `stylexBabelPlugin.withOptions(...)`, for the Rust side via `transform(filename, code, normalizeRsOptions(...))`.
3. Compare `metadata.stylex` from both.

The input imports a theme file. Save this beside it as `vars.stylex.js`:

```js
import * as stylex from '@stylexjs/stylex';
export const sliderVars = stylex.defineVars({ slidesPerView: '4' });
```

Class and `--x` variable hashes below depend on our repo's file paths, so a
local run will produce different hashes — compare the *shape* of the Babel vs
SWC outputs, which is path-independent.

#### Input

```js
import * as stylex from '@stylexjs/stylex';
import { sliderVars } from '../vars.stylex.js';
export const styles = stylex.create({
    theme: {
        [sliderVars.slidesPerView]: {
            '@media (min-width: 1632px)': '7',
            '@media (min-width: 1232px) and (max-width: 1631px)': '6',
            '@media (min-width: 1056px) and (max-width: 1231px)': '5',
            '@media (min-width: 800px) and (max-width: 1055px)': '4',
            '@media (min-width: 512px) and (max-width: 799px)': '3',
            '@media (max-width: 511px)': '2',
            default: null,
        },
    },
});
```

`@stylexjs/babel-plugin` 0.19.0 output (rule metadata):

```
xv1oflf  [priority 201]
  ltr: @media ((not all) or (not all)) or ((not all) or ((min-width: 1632px))){.xv1oflf.xv1oflf{--x5pecha:7}}
xv58bbs  [priority 201]
  ltr: @media (not all) or ((min-width: 1232px) and (max-width: 1631px)){.xv58bbs.xv58bbs{--x5pecha:6}}
xnb6s65  [priority 201]
  ltr: @media (min-width: 1056px) and (max-width: 1231px){.xnb6s65.xnb6s65{--x5pecha:5}}
x12lyiru  [priority 201]
  ltr: @media (min-width: 800px) and (max-width: 1055px){.x12lyiru.x12lyiru{--x5pecha:4}}
x12au5gi  [priority 201]
  ltr: @media (min-width: 512px) and (max-width: 799px){.x12au5gi.x12au5gi{--x5pecha:3}}
xuhliyr  [priority 201]
  ltr: @media (max-width: 511px){.xuhliyr.xuhliyr{--x5pecha:2}}
```

`@stylexswc/rs-compiler` 0.18.4-rc.2 output:

```
xztkim3  [priority 201]
  ltr: @media (min-width: 1632px){.xztkim3.xztkim3{--x5pecha:7}}
x1v9aby8  [priority 201]
  ltr: @media (min-width: 1232px) and (max-width: 1631px){.x1v9aby8.x1v9aby8{--x5pecha:6}}
xnb6s65  [priority 201]
  ltr: @media (min-width: 1056px) and (max-width: 1231px){.xnb6s65.xnb6s65{--x5pecha:5}}
x12lyiru  [priority 201]
  ltr: @media (min-width: 800px) and (max-width: 1055px){.x12lyiru.x12lyiru{--x5pecha:4}}
x12au5gi  [priority 201]
  ltr: @media (min-width: 512px) and (max-width: 799px){.x12au5gi.x12au5gi{--x5pecha:3}}
xuhliyr  [priority 201]
  ltr: @media (max-width: 511px){.xuhliyr.xuhliyr{--x5pecha:2}}
```

### Provide link to a repository with a minimal reproduction case.

n/a — the inputs above are complete.

### Additional information

- [x] I have checked with the latest version

---

_Generated by [Claude Code](https://claude.ai/code)_
